File: backend/timebank/src/main/java/kookmin/software/capstone2023/timebank/infrastructure/client/kakao/model/GetUserInfoResponse.java
package kookmin.software.capstone2023.timebank.infrastructure.client.kakao.model;

import java.time.LocalDateTime;

public class GetUserInfoResponse {
    /**
     * 회원번호
     */
    private final Long id;

    /**
     * 카카오계정 정보
     */
    private final KakaoAccount kakaoAccount;

    /**
     * 서비스 연결일시 (UTC)
     */
    private final LocalDateTime connectedAt;

    public GetUserInfoResponse(Long id, KakaoAccount kakaoAccount, LocalDateTime connectedAt) {
        this.id = id;
        this.kakaoAccount = kakaoAccount;
        this.connectedAt = connectedAt;
    }

    public Long getId() {
        return id;
    }

    public KakaoAccount getKakaoAccount() {
        return kakaoAccount;
    }

    public LocalDateTime getConnectedAt() {
        return connectedAt;
    }
}

File: src/main/java/kookmin/software/capstone2023/timebank/infrastructure/client/kakao/model/GetUserInfoResponse.kt
package kookmin.software.capstone2023.timebank.infrastructure.client.kakao.model

import java.time.LocalDateTime

data class GetUserInfoResponse(
    /**
     * 회원번호
     */
    val id: Long,

    /**
     * 카카오계정 정보
     */
    val kakaoAccount: KakaoAccount?,

    /**
     * 서비스 연결일시 (UTC)
     */
    val connectedAt: LocalDateTime?,
)
